cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -1174,32 +963,15 @@ impl ABIMachineSpec for AArch64MachineDeps {
         s.nominal_sp_to_fp
     }
 
-    fn get_regs_clobbered_by_call(call_conv_of_callee: isa::CallConv) -> PRegSet {
-        let mut clobbers = DEFAULT_AAPCS_CLOBBERS;
-
-        if call_conv_of_callee.extends_baldrdash() {
-            // Every X-register except for x16, x17, x18 is
-            // caller-saved (clobbered by a call).
-            for i in 19..=28 {
-                clobbers.add(xreg_preg(i));
-            }
-            clobbers.add(vreg_preg(31));
-        }
-
-        clobbers
+    fn get_regs_clobbered_by_call(_call_conv_of_callee: isa::CallConv) -> PRegSet {
+        DEFAULT_AAPCS_CLOBBERS
     }
 
     fn get_ext_mode(
-        call_conv: isa::CallConv,
-        specified: ir::ArgumentExtension,
+        _call_conv: isa::CallConv,
+        _specified: ir::ArgumentExtension,
     ) -> ir::ArgumentExtension {
-        if call_conv.extends_baldrdash() {
-            // Baldrdash (SpiderMonkey) always extends args and return values to the full register.
-            specified
-        } else {
-            // No other supported ABI on AArch64 does so.
-            ir::ArgumentExtension::None
-        }
+        ir::ArgumentExtension::None
     }
```

Comment:
I opted to keep `get_ext_mode` because it's used in s390x currently and I didn't want to dig too deeply to see how it might or might not be inlinable. Good question though!